pkg/utils/pci_allocator.go

```diff
@@ -0,0 +1,83 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"os"
+	"path/filepath"
+)
+
+type PCIAllocation interface {
+	SaveAllocatedPCI(string, string) error
+	CleanAllocatedPCI(string) error
+	IsAllocated(string) error
+}
+
+type PCIAllocator struct {
+	dataDir string
+}
+
+func NewPCIAllocator(dataDir string) PCIAllocation {
+	return &PCIAllocator{dataDir: filepath.Join(dataDir, "pci")}
+}
+
+// SaveAllocatedPCI creates a file with the pci address as a name and the network namespace as the content
+// return error if the file was not created
+func (p *PCIAllocator) SaveAllocatedPCI(pciAddress, ns string) error {
+	if err := os.MkdirAll(p.dataDir, 0600); err != nil {
+		return fmt.Errorf("failed to create the sriov data directory(%q): %v", p.dataDir, err)
+	}
+
+	path := filepath.Join(p.dataDir, pciAddress)
+	err := os.WriteFile(path, []byte(ns), 0600)
+	if err != nil {
+		return fmt.Errorf("failed to write used PCI address lock file in the path(%q): %v", path, err)
+	}
+
+	return err
+}
+
+// CleanAllocatedPCI Remove the allocated PCI file
+// return error if the file doesn't exist
+func (p *PCIAllocator) CleanAllocatedPCI(pciAddress string) error {
+	path := filepath.Join(p.dataDir, pciAddress)
+	if err := os.Remove(path); err != nil {
+		return fmt.Errorf("error removing PCI address lock file %s: %v", path, err)
+	}
+	return nil
+}
+
+// IsAllocated checks if the PCI address file exist
+// if it exists we also check the network namespace still exist if not we clean the allocation
+// The function will return an error if the pci is still allocated to a running pod
+func (p *PCIAllocator) IsAllocated(pciAddress string) error {
```

Comment:
the prefix Is in IsAllocated kinda makes you(me) expect to return a bool somewhere which i think it should.

id prefer IsAllocated returns bool and error, so the caller can understand if a device is allocated or not and if error occured.